src/main/java/com/gmail/goosius/siegewar/command/SiegeWarAdminCommand.java

```diff
@@ -226,30 +226,48 @@ private void parseSiegeWarImmunityCommand(CommandSender sender, String[] args) {
-			TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
-			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
-
-		} else if (args.length >= 3 && args[0].equalsIgnoreCase("nation")) {
+			if (args[2].equalsIgnoreCase("permanent")) {
+				TownMetaDataController.setSiegeImmunityEndTime(town, -1l);
+				TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_swa_town_unsiegeable_set", args[1]));
+				Messaging.sendMsg(sender, Translation.of("msg_swa_town_unsiegeable_set", args[1])); //this one
+			} else {
+				long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+				TownMetaDataController.setSiegeImmunityEndTime(town, System.currentTimeMillis() + durationMillis);
+				TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
+				Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
+			}
+		}
+		else if (args.length >= 3 && args[0].equalsIgnoreCase("nation")) {
 			//nation {nationname} {hours}
 			Nation nation = TownyUniverse.getInstance().getNation(args[1]);
 			if (nation == null) {
 				Messaging.sendErrorMsg(sender, Translation.of("msg_err_not_registered_1", args[1]));
 				return;
 			}
-			long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+			long endTime;
+			if (args[2].equalsIgnoreCase("permanent")) {
+				endTime = -1l;
+			} else {
+				long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+				endTime = System.currentTimeMillis() + durationMillis;
+			}
 			for (Town town : nation.getTowns()) {
-				TownMetaDataController.setSiegeImmunityEndTime(town, System.currentTimeMillis() + durationMillis);
+				TownMetaDataController.setSiegeImmunityEndTime(town, endTime);
 			}
 			TownyMessaging.sendPrefixedNationMessage(nation, Translation.of("msg_set_siege_immunities_nation", args[1], args[2]));
 			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_nation", args[1], args[2]));
 
 		} else if(args.length >= 2 && args[0].equalsIgnoreCase("alltowns")) {
 			//all towns
-			long durationMillis = (long)(Long.parseLong(args[1]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+			long endTime;
+			if (args[1].equalsIgnoreCase("permanent")) {
+				endTime = -1l;
+			} else {
+				long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
+				endTime = System.currentTimeMillis() + durationMillis;
+			}
 			for (Town town : new ArrayList<>(TownyUniverse.getInstance().getTowns()))  {
-				TownMetaDataController.setSiegeImmunityEndTime(town, System.currentTimeMillis() + durationMillis);
+				TownMetaDataController.setSiegeImmunityEndTime(town, endTime);
 			}
 			Messaging.sendGlobalMessage(Translation.of("msg_set_siege_immunities_all", args[1]));

```

Comment:
- If an admin selects the "permanent" option for all towns, they will see: "Siege immunities for all towns set to -1 hours."
- This is not a good user experience
- Please adjust. See last comment for an example of how to set up the lang strings for the 3 set-types.